### PR TITLE
feat(router): add forwarded_host options to support router behind a proxy

### PR DESCRIPTION
**Pull request by kunyan**
## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Review by sayak-sarkar — CHANGES_REQUESTED** [review]

File: packages/router/index.js [hunk 1]
```diff
@@ -134,8 +134,24 @@ let options = {
...
-// create the proxy
-let pathProxy = createProxyMiddleware(options);
+const pathProxy = (req, res, next) => {
+  const forwardedHost = config.get("forwarded_host");
+  const xForwaredHost = req.headers["x-forwarded-host"];
+  const host = req.headers["host"];
+
+  //
+  /**
+   * If we fount forwared_host in config, will use it as host
```
Comment (sayak-sarkar) [inline packages/router/index.js:144]:
```suggestion
   * If forwarded_host is found in config, use it as host
```

File: packages/router/index.js [hunk 2]
```diff
@@ -134,8 +134,24 @@ let options = {
...
+   * If x-forwared-host has some different value. will use it as host
```
Comment (sayak-sarkar) [inline packages/router/index.js:145]:
```suggestion
   * If x-forwarded-host has some different value, use it as host
```

**Review by sayak-sarkar — CHANGES_REQUESTED** [review]

**Comment by sayak-sarkar** [comment]
Hey @Jared-Sprague Can you please review this PR?

**Comment by sayak-sarkar** [comment]
Hey @Jared-Sprague Can you please review this PR as soon as possible? It's been in the pending review state for a month now.

**Review by Jared-Sprague — APPROVED** [review]
Looks good to me! Thanks @kunyan !!

**Review by sayak-sarkar — APPROVED** [review]

**Pull request merged**